gatorgrouper/utils/parse_arguments.py

```diff
@@ -98,9 +90,16 @@ def check_valid(args, students_list):
     """Verify the command-line arguments"""
     verified_arguments = False
     students_list_length = len(students_list)
-    if args.group_size > 1 and args.group_size <= students_list_length / 2:
+    if args.group_size is not None and args.group_size <= students_list_length / 2:
+        verified_arguments = True
+    if args.num_group is not None and args.num_group <= students_list_length / 2:
         verified_arguments = True
-    if args.num_group > 1 and args.group_size <= students_list_length / 2:
+    # pylint: disable=bad-continuation
+    if (
+        args.group_size is None
+        and args.num_group is None
+        and constants.DEFAULT_NUMGRP <= students_list_length / 2
+    ):
```

Comment:
In parse_argument.py if you set `required=True`, you don't need to check `args.group_size is None` and `args.num_group is None`here.

Reply:
For both of these comments, setting both of these arguments as a required will mess with the logic. Currently we check for which value (if any) was set explicitly by the user. If they set both, which one do we pick? How do they communicate that to us? What if the two values they pick are incompatible? These are questions that would come up if we set them to required. Ultimately the fact that these options can take the value of `None` is the centerpiece of the logic. I do think that it's ultimately not the best thing to do, but am unsure of what other way we should do it. If you have a plan or general design idea, we could talk about it and maybe prototype it in another branch?

Reply:
I'll explain that later when we meet.